app/code/Magento/InventoryShipping/view/adminhtml/ui_component/inventory_shipping_source_selection_form.xml

```diff
@@ -0,0 +1,217 @@
+        </item>
+        <item name="label" xsi:type="string" translate="true">Source Selection</item>
+        <item name="template" xsi:type="string">templates/form/collapsible</item>
+    </argument>
+    <settings>
+        <dataScope>data</dataScope>
+        <namespace>inventory_shipping_source_selection_form</namespace>
+        <deps>
+            <dep>inventory_shipping_source_selection_form.inventory_shipping_source_selection_form_data_source</dep>
+        </deps>
+        <buttons>
+            <button name="back">
+                <url path="sales/order/index"/>
+                <class>back</class>
+                <label translate="true">Back</label>
+            </button>
+            <button name="reset">
+                <class>reset</class>
+                <label translate="true">Reset</label>
+            </button>
+            <button name="save" >
+                <class>save primary</class>
+                <label translate="true">Create Shipment</label>
+            </button>
+            <button name="source_selection_algorithms" class="Magento\InventoryShipping\Ui\Component\Control\SourceSelection\AlgorithmSelectionButton"/>
+        </buttons>
+    </settings>
+    <dataSource name="inventory_shipping_source_selection_form_data_source" component="Magento_Ui/js/form/provider">
+        <settings>
+            <validateUrl path="*/*/validate"/>
+            <submitUrl path="admin/order_shipment/new"/>
+        </settings>
+        <dataProvider class="Magento\InventoryShipping\Ui\DataProvider\SourceSelectionDataProvider" name="inventory_shipping_source_selection_form_data_source">
+            <settings>
+                <requestFieldName>order_id</requestFieldName>
+                <primaryFieldName>order_id</primaryFieldName>
+            </settings>
+        </dataProvider>
+    </dataSource>
+    <fieldset name="general" sortOrder="10">
+        <settings>
+            <dataScope />
+            <label />
+            <collapsible>false</collapsible>
+            <opened>true</opened>
+        </settings>
+        <field name="order_id" formElement="hidden">
+            <settings>
+                <dataType>number</dataType>
+                <dataScope>order_id</dataScope>
+            </settings>
+        </field>
+        <field name="sourceCode" component="Magento_InventoryShipping/js/form/element/sourceCode" formElement="select" sortOrder="10">
+            <settings>
+                <validation>
+                    <rule name="required-entry" xsi:type="boolean">true</rule>
+                </validation>
+                <dataType>text</dataType>
+                <dataScope>sourceCode</dataScope>
+                <label translate="true">Select Source</label>
+            </settings>
+            <formElements>
+                <select>
+                    <settings>
+                        <caption translate="true">-- Please Select --</caption>
```

Comment:
Would be correct to specify options with class that provides this options, like here https://github.com/magento/magento2ce/blob/2.3-develop/app/code/Magento/Backend/view/adminhtml/ui_component/design_config_form.xml#L23. And then you can remove custom JS component that takes data from provider.